code/game/objects/effects/overlays.dm

```diff
@@ -21,10 +21,17 @@
 	name="beam"
 	icon='icons/effects/beam.dmi'
 	icon_state="b_beam"
+	layer = 13
+	mouse_opacity = 0
 	var/tmp/atom/BeamSource
-	New()
-		..()
-		spawn(10) qdel(src)
+
+/obj/effect/overlay/beam/New(turf/loc, var/lifetime = 10)

```

Comment:
Spawn is already a form of sleep, except it's more like a sleep that spools up a second process